shapeshift/web
# feat(mobile): add import via ShapeShift login

## Description

In the mobile app, provide an option to import a legacy ShapeShift wallet via email/password

This code re-uses the `LegacyLogin` code that exists for the `NativeWallet` provider. The existing code was modified to allow passing in of handler functions so that the mobile app and web app can save the imported wallet differently while re-using the same UI.

This does need to be tested in the web to make sure the import process doesn't have any regressions.

## Notice




## Pull Request Type


## Issue (if applicable)
See https://github.com/shapeshift/mobile-app/issues/43

## Risk
Medium

There is a risk that importing a legacy wallet got broken.
## Testing
This is easiest if you don't have any existing saved wallets

1. Make sure you have a valid ShapeShift email/password
2. Open the app in the browser
3. Go to Connect a Wallet
4. Go to ShapeShift
5. Click "Log In"
6. Enter your credentials
7. Check in "Saved Wallets" for your imported wallet

### Engineering
See "Testing"

To test in the mobile app,
1. in `mobile-app` pull the latest `main` and run `yarn android` or `yarn ios`
2. in `web` run `yarn dev:silent`
3. In the mobile app, shake the device (In Android emulator you can use the Device Position sensors to simulate a shake)
4. Select "Localhost" or "Localhost (Android)" for iOS or Android respectively
5. Follow the same steps above except select `ShapeShift Mobile` as the wallet type

### Operations
Test for regressions in importing a legacy wallet. See "Testing" section.

## Screenshots (if applicable)
![Screenshot_1663494542](https://user-images.githubusercontent.com/1958266/190896257-3f3e4e2c-6cc7-4904-8185-c0b4393b644f.png)
